ams/server/src/main/java/com/netease/arctic/server/optimizing/maintainer/IcebergTableMaintainer.java

```diff
@@ -136,9 +136,17 @@ public void cleanOrphanFiles(TableRuntime tableRuntime) {
 
     // refresh
     table.refresh();
-
-    // clear dangling delete files
-    cleanDanglingDeleteFiles();
+    Snapshot currentSnapshot = table.currentSnapshot();
+    java.util.Optional<String> total =
+        java.util.Optional.ofNullable(
+            currentSnapshot.summary().get(SnapshotSummary.TOTAL_DELETE_FILES_PROP));
+    if (total.isPresent()) {
+      // clear dangling delete files
+      cleanDanglingDeleteFiles();
+    } else {
+      LOG.debug(
+          "Table {} format version is v1,not need to clean dangling delete files", table.name());
```

Comment:
`Table {} does not have any delete files, so there is no need to clean dangling delete files` may be a more suitable phrasing.